gcloud/resource_manager/client.py

```diff
@@ -0,0 +1,298 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A Client for interacting with the Resource Manager API.
+
+Overview
+~~~~~~~~
+
+There are three main methods in the ``Client`` class:
+
+- :func:`gcloud.resource_manager.client.Client.list_projects`
+- :func:`gcloud.resource_manager.client.Client.project`
+- :func:`gcloud.resource_manager.client.Client.get_project`
+
+``project()`` versus ``get_project()``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The difference between ``project`` and ``get_project`` is subtle,
+so it might be worthwhile to make a quick distinction.
+
+If you want to simply "get a hold of a Project" object,
+but **don't** want to actually retrieve any metadata about that project
+(for example, when you want to create a new project, or delete a project
+in a single API request), ``project()`` is the best method to use::
+
+    >>> from gcloud import resource_manager
+    >>> client = resource_manager.Client()
+    >>> project = client.project('purple-spaceship-123')
+    >>> project.number is None
+    True
+
+The ``project`` referenced above has no extra metadata associated with it,
+however you can still operate on it (ie, ``project.create()`` or
+``project.delete()``).
+
+If you want to retrieve a project and all of it's metadata, the best method
+to use is ``get_project()``, which will return ``None`` if the project
+doesn't exist::
+
+    >>> from gcloud import resource_manager
+    >>> client = resource_manager.Client()
+    >>> project = client.get_project('purple-spaceship-123')
+    >>> project.number is None
+    False
+    >>> project = client.get_project('doesnt-exist')
+    >>> project is None
+    True
+"""
+
+
+from gcloud.credentials import get_credentials
+from gcloud.credentials import get_for_service_account_json
+from gcloud.credentials import get_for_service_account_p12
+from gcloud.exceptions import NotFound
+from gcloud.resource_manager.connection import Connection
+from gcloud.resource_manager.iterator import ProjectIterator
+from gcloud.resource_manager.project import Project
+
+
+class Client(object):
+    """Client to bundle configuration needed for API requests.
+
+    See
+    https://cloud.google.com/resource-manager/reference/rest/
+    for more information on this API.
+
+    Automatically get credentials::
+
+        >>> from gcloud.resource_manager import Client
+        >>> client = Client()
+
+    .. note::
+
+        Chances are you don't want to use the arguments in the constructor
+        but instead want one of the factory methods (like
+        :func:`gcloud.resource_manager.client.Client.from_service_account_json`
+        or similar).
+
+        Even more likely is that you want to use the Cloud SDK to get
+        credentials for these API calls (that is, run ``gcloud auth login``).
+
+    :type credentials: :class:`oauth2client.client.OAuth2Credentials` or
+                       :class:`NoneType`
+
```

Comment:
Oh ! You meant between type and param -- not between arguments... right ? In which case, sorry about that. Yes. Fixing now.